src/UCTSearch.cpp

```diff
@@ -615,13 +615,16 @@ void UCTSearch::ponder() {
     for (int i = 1; i < cpus; i++) {
         tg.add_task(UCTWorker(m_rootstate, this, m_root.get()));
     }
+    auto keeprunning = true;
     do {
         auto currstate = std::make_unique<GameState>(m_rootstate);
         auto result = play_simulation(*currstate, m_root.get());
         if (result.valid()) {
             increment_playouts();
         }
-    } while(!Utils::input_pending() && is_running());
+        keeprunning  = is_running();
+        keeprunning &= !stop_thinking(0, 1);
```

Comment:
Whoops, nice catch. It was definitely intended pondering should keep accurate --visits.

Reply:
Do we still need the m_maxvisits check inside if_running() that if we call stop_thinking in the same code path?

The only ones who don't call it now are the search threads but they will abort when the main thread does. So I think we can split the responsibility up, and have is_running just check the flag.

Reply:
I think is_running was changed recently. Now it only checks the flag and also MAX_TREE_SIZE, so I think it is close to what you are saying. We could move the MAX_TREE_SIZE check into stop_thinking also. There might be some weird spinning of the workers if the main search is busy expanding a NN eval, I guess it would be ok. What do you think?

Reply:
I'd make the change, I think it makes the code more sensible.